ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -140,9 +318,21 @@ public int compareTo(StoreKey o) {
 
     int result = version.compareTo(other.version);
     if (result == 0) {
-      result = partitionId.compareTo(other.partitionId);
+      result = flag.compareTo(other.flag);
       if (result == 0) {
-        result = uuid.compareTo(other.uuid);
+        result = datacenterId.compareTo(other.datacenterId);
+        if (result == 0) {
+          result = accountId.compareTo(other.accountId);
```

Comment:
Ok, after a second thought, it seems the current code could avoid some minor overhead if `result==0` at early checks. It will not execute the rest of checks.